json/src/de.rs

```diff
@@ -699,14 +699,14 @@ impl<'a, R: Read> de::Deserializer for &'a mut Deserializer<R> {
     fn deserialize_bytes<V>(self, visitor: V) -> Result<V::Value>
         where V: de::Visitor
     {
-        try!(self.parse_whitespace());
-
-        let next = try!(self.peek());
-        let next = try!(next.ok_or(self.peek_error(ErrorCode::ExpectedSomeString)));
+        if try!(self.parse_whitespace()) { // true if eof
+            return Err(self.peek_error(ErrorCode::EOFWhileParsingValue));
```

Comment:
yes, that is correct. I added it only to ensure the code is as similar as possible to `parse_value`, so correctness is easy to see. But it can be removed of course.